### PR TITLE
add getChannels

### PR DESCRIPTION
**Pull request by MariyaSin**

### CODE REVIEW

File: src/MessageService.Mappers/Models/WorkspaceInfoMapper.cs [hunk 1]
```diff
@@ -41,13 +41,13 @@ public WorkspaceInfo Map(DbWorkspace dbWorkspace, List<UserData> users)
         Description = dbWorkspace.Description,
         Image = image,
         CreatedAtUtc = dbWorkspace.CreatedAtUtc,
-        CreatedBy = _userInfoMapper.Map(creatorUserData),
+        CreatedBy = _userInfoMapper.Map(creatorUserData, null),
```
Comment (MariyaSin) [inline src/MessageService.Mappers/Models/WorkspaceInfoMapper.cs:44]: rework to do not use null

File: src/MessageService.Mappers/Models/WorkspaceInfoMapper.cs [hunk 2]
```diff
@@ -41,13 +41,13 @@ public WorkspaceInfo Map(DbWorkspace dbWorkspace, List<UserData> users)
...
         IsActive = dbWorkspace.IsActive,
         Channels = dbWorkspace.Channels?
           .Select(_channelInfoMapper.Map).ToList(),
         Users = users?
           .Where(u => dbWorkspace.Users.Any(wu => wu.UserId == u.Id))
-          .Select(_userInfoMapper.Map).ToList()
+          .Select(u => _userInfoMapper.Map(u, null)).ToList()
```
Comment (MariyaSin) [inline src/MessageService.Mappers/Models/WorkspaceInfoMapper.cs:50]: do not use null

**Review by Cheephot — APPROVED** [review]

**Review by VLiamin — APPROVED** [review]

File: src/MessageService.Business/Commands/Channels/GetChannelCommand.cs [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+      if (!dbChannel.Workspace.Users.Select(wu => wu.UserId).Contains(requestUserId)
+        || (dbChannel.IsPrivate
+          && !dbChannel.Users.Where(u => u.IsActive).Select(u => u.WorkspaceUser.UserId).Contains(requestUserId)))
+      {
+        return _responseCreator.CreateFailureResponse<ChannelInfo>(HttpStatusCode.Forbidden);
+      }
+
+      OperationResultResponse<ChannelInfo> response = new();
+
+
```
Comment (BelehovEgor) [inline src/MessageService.Business/Commands/Channels/GetChannelCommand.cs:160]: remove

File: src/MessageService.Business/Commands/Workspace/GetWorkspaceCommand.cs [hunk 4]
```diff
@@ -89,19 +144,24 @@ public async Task<OperationResultResponse<WorkspaceInfo>> ExecuteAsync(GetWorksp
...
       List<UserData> usersData = await GetUsersAsync(dbWorkspace.Users?.Select(u => u.UserId).ToList(), response.Errors);
 
-      List<Guid> imagesIds = new();
-
-      if (usersData is not null)
-      {
-        imagesIds.AddRange(usersData.Where(u => u.ImageId.HasValue).Select(u => u.ImageId.Value).ToList());
-      }
+      List<ImageInfo> imagesInfo = (await GetImagesAsync(
+        usersData?.Where(u => u.ImageId.HasValue).Select(u => u.ImageId.Value).ToList(),
```
Comment (BelehovEgor) [inline src/MessageService.Business/Commands/Workspace/GetWorkspaceCommand.cs:157]: add tab for 3 next time

File: src/MessageService.Data/ChannelRepository.cs [hunk 5]
```diff
@@ -27,5 +30,28 @@ public ChannelRepository(IDataProvider provider)
 
       return dbChannel.Id;
     }
+
+    public async Task<DbChannel> GetAsync(GetChannelFilter filter)
+    {
+      IQueryable<DbChannel> dbChannel = _provider.Channels.AsQueryable();
+
+      dbChannel = dbChannel.Include(c => c.Workspace).ThenInclude(w => w.Users);
```
Comment (BelehovEgor) [inline src/MessageService.Data/ChannelRepository.cs:38]: you can do it in line 36

File: src/MessageService.Models.Db/DbMessageFile.cs [hunk 6]
```diff
@@ -32,11 +32,13 @@ public void Configure(EntityTypeBuilder<DbMessageFile> builder)
 
       builder
         .HasOne(mf => mf.Message)
-        .WithMany(m => m.Files);
+        .WithMany(m => m.Files)
+        .HasForeignKey(mf => mf.MessageId);
```
Comment (BelehovEgor) [inline src/MessageService.Models.Db/DbMessageFile.cs:36]: you don't needed

File: src/MessageService.Models.Db/DbMessageFile.cs [hunk 7]
```diff
@@ -32,11 +32,13 @@ public void Configure(EntityTypeBuilder<DbMessageFile> builder)
...
 
       builder
         .HasOne(mf => mf.ThreadMessage)
-        .WithMany(m => m.Files);
+        .WithMany(tm => tm.Files)
+        .HasForeignKey(mf => mf.MessageId);
```
Comment (BelehovEgor) [inline src/MessageService.Models.Db/DbMessageFile.cs:41]: same

File: src/MessageService.Models.Db/DbMessageImage.cs [hunk 8]
```diff
@@ -32,11 +32,13 @@ public void Configure(EntityTypeBuilder<DbMessageImage> builder)
 
       builder
         .HasOne(mi => mi.Message)
-        .WithMany(m => m.Images);
+        .WithMany(m => m.Images)
+        .HasForeignKey(mi => mi.MessageId);
```
Comment (BelehovEgor) [inline src/MessageService.Models.Db/DbMessageImage.cs:36]: same

File: src/MessageService.Models.Db/DbMessageImage.cs [hunk 9]
```diff
@@ -32,11 +32,13 @@ public void Configure(EntityTypeBuilder<DbMessageImage> builder)
...
 
       builder
-        .HasOne(mf => mf.ThreadMessage)
-        .WithMany(m => m.Images);
+        .HasOne(mi => mi.ThreadMessage)
+        .WithMany(tm => tm.Images)
+        .HasForeignKey(mi => mi.MessageId);
```
Comment (BelehovEgor) [inline src/MessageService.Models.Db/DbMessageImage.cs:41]: same

File: src/MessageService.Models.Dto/Configurations/RabbitMqConfig.cs [hunk 10]
```diff
@@ -1,19 +1,24 @@
...
     public string CreateWorkspaceEndpoint { get; set; }
-    public string SendEmailEndpoint { get; set; }
+
 
     [AutoInjectRequest(typeof(IGetUsersDataRequest))]
     public string GetUsersDataEndpoint { get; set; }
 
     [AutoInjectRequest(typeof(ICheckUsersExistence))]
     public string CheckUsersExistenceEndpoint { get; set; }
+
```
Comment (BelehovEgor) [inline src/MessageService.Models.Dto/Configurations/RabbitMqConfig.cs:19]: remove

File: src/MessageService.Models.Dto/Configurations/RabbitMqConfig.cs [hunk 11]
```diff
@@ -1,19 +1,24 @@
...
+using LT.DigitalOffice.Models.Broker.Requests.Image;
 using LT.DigitalOffice.Models.Broker.Requests.User;
 
 namespace LT.DigitalOffice.MessageService.Models.Dto.Configurations
 {
   public class RabbitMqConfig : BaseRabbitMqConfig
   {
     public string CreateWorkspaceEndpoint { get; set; }
-    public string SendEmailEndpoint { get; set; }
+
```
Comment (BelehovEgor) [inline src/MessageService.Models.Dto/Configurations/RabbitMqConfig.cs:12]: remove

File: src/MessageService.Models.Dto/Filtres/GetChannelFilter.cs [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+using Microsoft.AspNetCore.Mvc;
+
+namespace LT.DigitalOffice.MessageService.Models.Dto.Filtres
+{
+  public class GetChannelFilter
+  {
+    [FromQuery(Name = "channelid")]
+    public Guid ChannelId { get; set; }
+
+    [FromQuery(Name = "skipmessages")]
```
Comment (BelehovEgor) [inline src/MessageService.Models.Dto/Filtres/GetChannelFilter.cs:11]: count*?

File: src/MessageService.Models.Db/DbMessageFile.cs (hunk 6 above)
Comment (MariyaSin) [reply src/MessageService.Models.Db/DbMessageFile.cs:36]: now i can not remove it

File: src/MessageService.Models.Db/DbMessageFile.cs (hunk 7 above)
Comment (MariyaSin) [reply src/MessageService.Models.Db/DbMessageFile.cs:41]: same

File: src/MessageService.Models.Db/DbMessageImage.cs (hunk 8 above)
Comment (MariyaSin) [reply src/MessageService.Models.Db/DbMessageImage.cs:36]: same

File: src/MessageService.Models.Db/DbMessageImage.cs (hunk 9 above)
Comment (MariyaSin) [reply src/MessageService.Models.Db/DbMessageImage.cs:41]: same

File: src/MessageService.Business/Commands/Channels/GetChannelCommand.cs [hunk 13]
```diff
@@ -0,0 +1,190 @@
...
+        return null;
+      }
+
+      try
+      {
+        Response<IOperationResult<IGetImagesResponse>> response =
+          await _rcGetImages.GetResponse<IOperationResult<IGetImagesResponse>>(
+            IGetImagesRequest.CreateObj(imagesIds, ImageSource.User));
+
+        if (response.Message.IsSuccess)
```
Comment (ndoriomedov) [inline src/MessageService.Business/Commands/Channels/GetChannelCommand.cs:93]: add !(body is null) check

**Review by ndoriomedov — APPROVED** [review]

**Review by asyagraf — APPROVED** [review]

**Review by actimellka — APPROVED** [review]

**Pull request merged**